src/dotnet-scaffolding/Microsoft.DotNet.Scaffolding.Core/Builder/ScaffoldRunnerBuilder.cs

```diff
@@ -72,13 +75,42 @@ private void AddDefaultServices()
 
     private void AddCoreServices()
     {
-        Services.AddLogging();
-        Logging.AddCleanConsoleFormatter();
+        //get some logging properties from environment variables
+        var isVerboseEnabled = Environment.GetEnvironmentVariable(ScaffolderConstants.ENABLE_VERBOSE_LOGGING)?.Equals("true", StringComparison.OrdinalIgnoreCase) == true;
+        var isLogToFileEnabled = Environment.GetEnvironmentVariable(ScaffolderConstants.LOG_TO_FILE)?.Equals("true", StringComparison.OrdinalIgnoreCase) == true; ;
+        // Configure Serilog Logger
+        var loggerConfig = new LoggerConfiguration().WriteTo.Sink(new AnsiConsoleSink());
+        loggerConfig.MinimumLevel.Information();
+        if(isVerboseEnabled)
+        {
+            loggerConfig.MinimumLevel.Verbose();
+        }
+
+        if (isLogToFileEnabled)
+        {
+            var currentDirectory = Directory.GetCurrentDirectory();
+            if (!string.IsNullOrEmpty(currentDirectory))
+            {
+                var filePath = $"dotnet-scaffold-{DateTime.UtcNow:yyyy-MM-dd_HH-mm}.log";
+                var logPath = StringUtil.GetUniqueFilePath(Path.Combine(currentDirectory, _defaultLogFolder, filePath));                
+                loggerConfig.WriteTo.File(logPath, rollingInterval: RollingInterval.Infinite);
+            }
+        }
+
+        Log.Logger = loggerConfig.CreateLogger();
```

Comment:
Out of curiosity, did you try the configuration constructs mentioned here? https://github.com/serilog/serilog-extensions-hosting ? I ask just because I like the inline variant of it that allows for not touching the static `Log.Logger`. Though i'm not entirely sure the end result is the same - its technically a different package (`Serilog.Extensions.Hosting` vs `Serilog.Extensions.Logging`). Though the note on the [Logging readme](https://github.com/serilog/serilog-extensions-logging?tab=readme-ov-file) says 

> Non-web .NET Core Instructions
Non-web .NET Core applications should prefer [Serilog.Extensions.Hosting](https://github.com/serilog/serilog-extensions-hosting) and UseSerilog() instead.

Reply:
oh yeah i missed this I think. I'll play with it

Reply:
actually didn't need the other package at all, builder.AddSerilog has an overload to add the created LoggerConfiguration, looks like `builder.AddSerilog(loggerConfig.CreateLogger(), dispose: true);` now. Tested and everything is good.